cli/command/trust/signer_add.go

```diff
@@ -44,9 +45,13 @@ func newSignerAddCommand(dockerCli command.Cli) *cobra.Command {
 	return cmd
 }
 
+var validSignerName = regexp.MustCompile(`^[a-zA-Z0-9\_]+[a-zA-Z0-9\_\-]*$`).MatchString
+
 func addSigner(cli command.Cli, options *signerAddOptions) error {
 	signerName := options.signer
-
+	if !validKeyName(signerName) {
+		return fmt.Errorf("signer name \"%s\" must not contain special characters", signerName)
```

Comment:
let's also update this message to reflect the alphabetical lower case restriction